cabin-sorting/api/sortingData.js

```diff
@@ -0,0 +1,96 @@
+import path from "path";
+import fs from "fs";
+import { client, connectToDatabase } from "./mongodb";
+
+let validatedHackers;
+const validHackers = [];
+
+// Get all valid hacker data from the applicant_data database and convert to json contents
+async function grabFromDatabase() {
+  const applicantData = await connectToDatabase("applicant_data");
+
+  try {
+    // Query for hackers whose application status is submitted and who has a
+    // 'Post Acceptance Response' section
+    const query = {
+      applicationStatus: "Submitted",
+      postAcceptanceResponses: { $exists: true },
+      isAdmin: false,
+      rsvpStatus: "Confirmed",
+      "postAcceptanceResponses.swag": { $exists: false },
+      "postAcceptanceResponses.club": { $exists: true },
+    };
+
+    // Find all valid hackers that have answers to the cabin questions
+    const hackerDataCursor = applicantData
```

Comment:
cursor?

Reply:
A cursor is a type that originates from MongoDb and gathers all the information from a document. From the cursor you can then iterate over the items in the Document. We added a new comment in the code to say something along these lines <3